scheds/rust/scx_p2dq/src/bpf/main.bpf.c

```diff
@@ -655,44 +676,29 @@ void BPF_STRUCT_OPS(p2dq_dispatch, s32 cpu, struct task_struct *prev)
 		return;
 	}
 
-	// In the dispatch path we first figure out the load factor for each dsq.
-	s32 nr_queued = 0;
-	int max_dsq = 0;
-	u64 dsq_load = 0;
-	u64 max_load = 0;
+	u64 max_vtime = llcx->vtime;
 	bpf_for(i, 0, nr_dsqs_per_llc) {
-		nr_queued = scx_bpf_dsq_nr_queued(*MEMBER_VPTR(cpuc->dsqs, [i]));
-		if (nr_queued <= 0)
-			continue;
-
-		dsq_load = i <= nr_queued ? nr_queued : nr_queued / i;
-		if (dsq_load > max_load) {
-			max_load = dsq_load;
-			max_dsq = i;
+		if (llcx->dsq_max_vtime[i] < max_vtime) {
```

Comment:
Isn't this the other way around? (> instead of <)

Reply:
The naming is kind of bad, it's the max vtime for that DSQ, but it should try to consume from the DSQ with the vtime that is most behind is the thinking.